core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -448,4 +449,18 @@ public String toString() {
     }
     return sb.toString();
   }
+
+  /**
+   * Gets the path component of the {@link AlluxioURI}.
+   *
+   * @return the absolute path
+   */
+  public String getAbsolutePath() {
+    if (mUri.getAuthority() != null) {
+      return PathUtils.concatPath(System.getProperty("user.dir"),
+              mUri.getAuthority().equals(".") ? mUri.getPath() : mUri.getAuthority());
```

Comment:
I think this PR is still valuable for `copyToLocal` work with absolute-path directories work. You are right that we need to change the `URI.Factory` parsing methods to be able to resolve relative paths with the `file://` scheme into absolute paths. I think that's the simplest path forward for supporting all relative paths, which is really important for making the `cp` command easy to use